CONTRIBUTING.md

```diff
@@ -55,6 +55,14 @@ git checkout -b cool-new-feature
 
 4\. When you are ready, submit a pull request into the composer repository! If merged, we'll reach out to send you some free swag :)
 
+## Pull request guidelines
+
+We have some rough guidelines that will make your PR easier to review and more likely to get smoothly merged.
+* Self-contained title and description. Please include a descriptive title and description. These should allow someone to understand the contents of the PR without looking at the code.
+* If the PR affects output that is displayed to a user of Composer (e.g. console logging or experiment tracker reporting), please include screenshots showing what the new output looks like. UX is important!
+* Include tests. If you are fixing a bug, please add a test that would've caught the bug. If you are adding a new feature, please add unit tests that test the various components of the feature, and also a test that tests the full functionality of the feature.
+* `pre-commit` should help you handle formatting and type checking, but please do make sure you have it installed as described [above](#prerequisites).
+
```

Comment:
I'm thinking about enabling this soon:
https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

Just mentioning here because we can start using labels as well and it will greatly simplify creating release notes.

Reply:
Also we should mention what is must and nice-to-have out of these. Basically we don't want to give a big list of things to do to external contributors.

Reply:
I consider all of these things essentially required (except running daily tests), but I will add some language up front saying that any uncertainty/difficulty with any of these should not deter from opening a PR and we are happy to help